### PR TITLE
docs: add install karmada docs

### LINKED ISSUE #30: Add document directory
**Please provide an in-depth description of the question you have**:

I'm researching kurator and kurator looks like a good fit for us. I can output some installation or other documentation during the research process.

I don't know if the community has a documentation plan.

**What do you think about this question?**:

I want to put the documentation in the `docs` directory.

**Environment**:
- Kurator version:
- Others:

### PR DESCRIPTION
**Pull request by prodanlabs**
Signed-off-by: prodan <pengshihaoren@gmail.com>

**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #30

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Karmada
```
Comment (hzxuzhonghu) [inline docs/README.md:1]: ?

Comment (prodanlabs) [reply docs/README.md:1]: sorry my fault.

File: docs/install-components/karmada.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+Deploy a kubernetes cluster using kurator's scripts. This script will create three clusters for you, one is used to host Karmada control plane and the other two will be joined as member clusters.
+```bash
+git clone https://github.com/kurator-dev/kurator.git
+cd kurator
+hack/local-dev-setup.sh
+```
+
+### KinD
+
+install docker
```
Comment (zirain) [inline docs/install-components/karmada.md:33]: I would prefer to link to an official doc if exists.

File: docs/install-components/karmada.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+## Deploy Karmada
+
+Compile `kurator` from source
+```bash
+git clone https://github.com/kurator-dev/kurator.git
+make kurator
+```
+
+Download `kurator` from the Releases page
```
Comment (zirain) [inline docs/install-components/karmada.md:145]: should we just remove this?

File: docs/install-components/karmada.md [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+
+karmada installation parameters can be set with `--set`, e.g.
+```
+./kurator install karmada --set karmada-data=/etc/Karmada-test --set port=32222 --kubeconfig .kube/config
+```
+
+## Add kubernetes cluster to karmada control plane
+
+```bash
+kurator join karmada member1 \
```
Comment (zirain) [inline docs/install-components/karmada.md:164]: this should be `./kurator`, same as L158?

File: docs/install-components/karmada.md [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+# Install Karmada
+
+The documentation uses `Ubuntu 20.04.4 LTS` as an example.
+
+## Prerequisites
+
+Installing Karmada requires a kubernetes cluster. If you do not have a kubernetes cluster, you can choose any of the following methods to deploy the kubernetes cluster.
+
+### Script
+Deploy a kubernetes cluster using kurator's scripts. This script will create three clusters for you, one is used to host Karmada control plane and the other two will be joined as member clusters.
```
Comment (zirain) [inline docs/install-components/karmada.md:24]: Deploy kubernetes clusters using kurator's script

File: docs/install-components/karmada.md (hunk 4 above)
Comment (zirain) [reply docs/install-components/karmada.md:164]: maybe we can update README and then link to it.

File: docs/install-components/karmada.md (hunk 3 above)
Comment (prodanlabs) [reply docs/install-components/karmada.md:145]: I personally think that kurator will release binaries of Releases version in the future, we can't let each user compile kurator by himself.

Do you mean to wait for the release version to add it?

File: docs/install-components/karmada.md (hunk 2 above)
Comment (prodanlabs) [reply docs/install-components/karmada.md]: done

File: docs/install-components/karmada.md (hunk 5 above)
Comment (prodanlabs) [reply docs/install-components/karmada.md:25]: done

File: docs/install-components/karmada.md (hunk 4 above)
Comment (prodanlabs) [reply docs/install-components/karmada.md]: > maybe we can update README and then link to it.

Is `/docs/README.md` ?

File: docs/install-components/karmada.md (hunk 3 above)
Comment (zirain) [reply docs/install-components/karmada.md:130]: > Do you mean to wait for the release version to add it?

yes

File: docs/install-components/karmada.md (hunk 4 above)
Comment (zirain) [reply docs/install-components/karmada.md]: sorry, I put my comment in wrong line.

File: docs/install-components/karmada.md [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+> `kubectl taint node ${NODE_NAME} node-role.kubernetes.io/control-plane-`
+> 
+> `kubectl taint node ${NODE_NAME} node-role.kubernetes.io/master-`
+
+
+For more instructions, please refer to [kubeadm](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm)
+
+## Deploy Karmada
+
+Compile `kurator` from source
```
Comment (zirain) [inline docs/install-components/karmada.md:124]: maybe we should link this to the README.md of repo, which I think will keep updating.

another option is I will take a look about the doc infra, want to find a way that we can insert snippet like istio.io do.

File: docs/install-components/karmada.md (hunk 3 above)
Comment (prodanlabs) [reply docs/install-components/karmada.md]: done

File: docs/install-components/karmada.md (hunk 6 above)
Comment (prodanlabs) [reply docs/install-components/karmada.md:124]: Sorry, I don't quite understand what you mean.

Personally I think the repo's README.md is quickstart, and the installation documentation should be more detailed than quickstart.

Comment (prodanlabs) [reply docs/install-components/karmada.md:124]: However, about compiling kurator from source or downloading kurator, it really shouldn't appear here.

Comment (zirain) [reply docs/install-components/karmada.md:124]: After thinking, let's put this section here at this stage.

**Pull request merged**